### PR TITLE
Use gz- projects on Homebrew CI

### PR DESCRIPTION
**Pull request by chapulina**
* Requires https://github.com/osrf/homebrew-simulation/pull/1984
* The intention is to fix CI for https://github.com/gazebosim/gz-cmake/pull/296

Test build using this PR together with the 2 above: [Build Status](https://build.osrfoundation.org/job/ignition_cmake-ci-pr_any-homebrew-amd64/911/)

---

Ah had to rebuild the DSL with this branch: [Build Status](https://build.osrfoundation.org/job/_dsl_ignition/1340/)

Build after rebuilding the DSL: [Build Status](https://build.osrfoundation.org/job/ignition_cmake-ci-pr_any-homebrew-amd64/912/)

Annnd I built the wrong DSL before, I think this should do it: [Build Status](https://build.osrfoundation.org/job/_dsl_ignition_collection/570/)

New new build: [Build Status](https://build.osrfoundation.org/job/ignition_cmake-ci-pr_any-homebrew-amd64/913/)

---

After [465385a](https://github.com/gazebo-tooling/release-tools/pull/778/commits/465385af1183a0065042ff631aa4bbfe9797a8e4)

Another DSL  [Build Status](https://build.osrfoundation.org/job/_dsl_ignition/1341/)

New new new build: [Build Status](https://build.osrfoundation.org/job/ignition_cmake-ci-pr_any-homebrew-amd64/914/)

---

After [ef6148b](https://github.com/gazebo-tooling/release-tools/pull/778/commits/ef6148b68362ae68d7f9c963f28b4b18ce7886df)

DSLs

* [Build Status](https://build.osrfoundation.org/job/_dsl_ignition/1342/)
* [Build Status](https://build.osrfoundation.org/job/_dsl_ignition_collection/571/)

Build: [Build Status](https://build.osrfoundation.org/job/ignition_cmake-ci-pr_any-homebrew-amd64/915/)

### CODE REVIEW

**Comment by chapulina** [comment]
Success!

```
+ brew install gz-cmake4 --only-dependencies
...
==> Installing ignition-cmake3 from osrf/simulation
==> Installing dependencies for osrf/simulation/ignition-cmake3: pkg-config
==> Installing osrf/simulation/ignition-cmake3 dependency: pkg-config
==> Pouring pkg-config--0.29.2_3.big_sur.bottle.tar.gz
```

File: jenkins-scripts/lib/project-default-devel-homebrew-amd64.bash [hunk 1]
```diff
@@ -13,10 +13,10 @@ export HOMEBREW_MAKE_JOBS=${MAKE_JOBS}
 PROJECT=$1 # project will have the major version included (ex gazebo2)
 PROJECT_ARGS=${2}
 
-# In ignition projects, the name of the repo and the formula does not match
+# TODO(chapulina) Use gz path instead of legacy ign
```
Comment (j-rivero) [inline jenkins-scripts/lib/project-default-devel-homebrew-amd64.bash:16]: Any particular reason not to use the new gz- and use the legacy instead?

**Comment by j-rivero** [comment]
The CI includes the testing of bottles so we are going to need to modify [the bottle name var in DSL](https://github.com/gazebo-tooling/release-tools/pull/778/files#diff-1269f38223f9cd477137c0a99b7e9ff46f8eeb3b09300d2b245ec8f9381172f5R616) when new `gz-` bottles are created.  We can create a different PR for this if you prefer to keep this one small and go fast.

**Review by j-rivero — APPROVED** [review]
Minor comment about not using the legacy mechanism but looks good in either way.

File: jenkins-scripts/lib/project-default-devel-homebrew-amd64.bash (hunk 1 above)
Comment (chapulina) [reply jenkins-scripts/lib/project-default-devel-homebrew-amd64.bash:16]: That was my first approach, and the test run showed that more things would need to be changed, so I though of keeping this PR to the strictly necessary for now.

**Comment by chapulina** [comment]
> The CI includes the testing of bottles so we are going to need to modify [the bottle name var in DSL](https://github.com/gazebo-tooling/release-tools/pull/778/files#diff-1269f38223f9cd477137c0a99b7e9ff46f8eeb3b09300d2b245ec8f9381172f5R616) when new gz- bottles are created.

I hadn't realized that. So is merging this PR going to break anything? I thought I was only affecting these CI jobs:

* `ignition_${ign_sw}-ci-pr_any-homebrew-amd64`
* `ignition_${ign_sw}-ci-${branch}-homebrew-amd64`
* `ignition_${ign_collection_name}-ci-main-homebrew-amd64`

**Comment by j-rivero** [comment]
> I hadn't realized that. So is merging this PR going to break anything? I thought I was only affecting these CI jobs:

It is not going to break anything, I was reviewing the status of other things affected by migration as part of Homebrew CI.

**Pull request merged**